clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3177,25 +3177,32 @@ static void handleWorkGroupSize(Sema &S, Decl *D, const ParsedAttr &AL) {
 
   ASTContext &Ctx = S.getASTContext();
 
-  if (!XDimExpr->isValueDependent() && !YDimExpr->isValueDependent() &&
-      !ZDimExpr->isValueDependent()) {
-    llvm::APSInt XDimVal, YDimVal, ZDimVal;
-    ExprResult XDim = S.VerifyIntegerConstantExpression(XDimExpr, &XDimVal);
-    ExprResult YDim = S.VerifyIntegerConstantExpression(YDimExpr, &YDimVal);
-    ExprResult ZDim = S.VerifyIntegerConstantExpression(ZDimExpr, &ZDimVal);
+  bool YDimExprIsDereferencable = YDimExpr && (!YDimExpr->isValueDependent());
+  bool ZDimExprIsDereferencable = ZDimExpr && (!ZDimExpr->isValueDependent());
```

Comment:
```suggestion
  bool YDimExprIsDereferencable = YDimExpr && !YDimExpr->isValueDependent();
  bool ZDimExprIsDereferencable = ZDimExpr && !ZDimExpr->isValueDependent();
```
Parens don't change the order of evaluation here, so they can be removed.

Reply:
Ok.